test/integration/adoptable_pet_show_test.rb

```diff
@@ -28,6 +29,12 @@ class AdoptablePetShowTest < ActionDispatch::IntegrationTest
       assert_response :success
     end
 
+    should "see a pet with a current foster" do
```

Comment:
This scenario is currently not possible in Main because the policy is rejecting pets with **any** matches, including those that are fosters. So, this is a needed fix.